diff --git a/tools/chapel-py/setup.py b/tools/chapel-py/setup.py
index bee452790c..52bb922534 100644
--- a/tools/chapel-py/setup.py
+++ b/tools/chapel-py/setup.py
@@ -46,7 +46,24 @@ host_cc = str(chpl_variables.get("CHPL_HOST_CC"))
 host_cxx = str(chpl_variables.get("CHPL_HOST_CXX"))

 host_bin_subdir = str(chpl_variables.get("CHPL_HOST_BIN_SUBDIR"))
-chpl_lib_path = os.path.join(chpl_home, "lib", "compiler", host_bin_subdir)
+
+# Add the chplenv dir to the python path.
+chplenv_dir = os.path.join(os.path.dirname(__file__), '../../util/chplenv')
+sys.path.insert(0, os.path.abspath(chplenv_dir))
+
+import chpl_home_utils
+
+# construct the chpl_lib_path from chpl_home, or use the configured-prefix if it exists
+if chpl_home_utils.get_prefix_install_prefix() is None:
+    chpl_lib_path = os.path.join(chpl_home, "lib", "compiler", host_bin_subdir)
+else:
+    chpl_lib_path = os.path.join(
+        chpl_home_utils.get_prefix_install_prefix(),
+        "lib",
+        "chapel",
+        chpl_home_utils.get_version_from_install(),
+        "compiler"
+    )

 CXXFLAGS = []
 if have_llvm and have_llvm != "none":
@@ -64,9 +81,8 @@ CXXFLAGS += ["-std=c++17", "-I{}/frontend/include".format(chpl_home)]
 LDFLAGS = []
 LDFLAGS += [
     "-L{}".format(chpl_lib_path),
+    "-Wl,-rpath,{}".format(chpl_lib_path),
     "-lChplFrontendShared",
-    "-Wl,-rpath",
-    chpl_lib_path,
 ]

 if str(chpl_variables.get("CHPL_SANITIZE")) == "address":
